### PR TITLE
Project: Library JS

### PR DESCRIPTION
**Pull request by LamiaSristy**
## This project is a simple library where a user can add and delete books.

- Here we have practiced JS Objects and Object contractors.
- DOM Manipulation with JavaScript.
- Use of JavaScript functions and loops

## The project includes the following files:
- index.HTML
- styles.css
- app.js.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+function Book(title, author, pages, isread) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.isread = isread;
+}
+
+function addBookToLibrary(book) {
+  myLibrary.push(book);
+}
```
Comment (addod19) [inline app.js]: Kindly change the syntax to ES6. For example `const addBookToLibrary = (book) => {}`

File: app.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  }
+  // toggle btn for isRead
+  readBtn.addEventListener('click', () => {
+    book.isread = !book.isread;
+    readBtn.textContent = book.isread ? 'Read' : 'NotRead';
+    readBtn.classList.toggle('readbtn');
+    readBtn.classList.toggle('notreadbtn');
+  });
+  return readBtn;
+}
```
Comment (addod19) [inline app.js]: Kindly change the syntax to ES6. For example `const createIsReadBtn = (book) => {}`

File: app.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    const author = document.createElement('p');
+    const pages = document.createElement('p');
+
+    // Data insertion
+    title.textContent = book.title;
+    author.textContent = `Author: ${book.author}`;
+    pages.textContent = `No.of pages: ${book.pages}`;
+
+    // delete a book Btn
+    const deleteBtn = document.createElement('button');
```
Comment (addod19) [inline app.js]: Kindly change the syntax to ES6. For example `const displayLibrary = array  => {}`

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @LamiaSristy and @talhawaqar, great job on this project :+1: However there are a few things we need to address before proceeding :slightly_smiling_face: Kindly check the comments below: :smile: 
- Kindly add a package.json file add the scripts to it
- Kindly separate DOM manipulation from the application logic. We can create a file for our application logic and a different file for our DOM manipulation. When we have all in one big file, it becomes hard to test
- Do not use `window.alert()` or `window.confirm()` but instead use custom modal. Let us fix it
- Kindly use object destructuring to get the values from an object. This way you can avoid repeating a lot of code


[OPTIONAL]
- Kindly implement data persistence by saving, retrieving, and updating the library books using the localstorage API.
- Kindly make the app responsive

Once these are fixed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: :woman_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: app.js (hunk 2 above)
Comment (LamiaSristy) [reply app.js]: This is our very first project on JS module. ES6 is yet to come. So I didn't add ES6 function stylings.

File: app.js (hunk 1 above)
Comment (LamiaSristy) [reply app.js]: This was given on the project spec. I was just following the instructions.
ss3

File: app.js (hunk 3 above)
Comment (LamiaSristy) [reply app.js]: ES6 and Webpack will be added when I will be on the next step. Did you read the requirements?

Comment (LamiaSristy) [reply app.js]: sss

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi **TEAM**, most of the comments were not addressed, below is a screenshot of the requirements, kindly check it and work on it accordingly

st
- You have removed the `window.alert()` but you did not implement the custom modal. Kindly check the screenshot and implement the changes


Once this is fixed, kindly submit another code review :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Review by Forison — APPROVED** [review]
## Approved✔️
#### Your code looks clean, kindly go ahead and merge the branch and proceed to the next task🤗.
> I saw no `package.json`, but I saw a `package-lock.json`, you also did not have its path in your gitignore, I am wondering did you use any external package in the application(I can see you did not), if no then why have `package-lock.json`, I am just curious🤔.
#### Happy coding
Addo Forison

**Pull request merged**